File: 2_Kernel/JavaKernel/src/main/java/com/seagosoft/tws/data/TickByTickBidAskData.java
package com.seagosoft.tws.data;

import com.ib.client.Decimal;
import com.ib.client.TickAttribBidAsk;

public class TickByTickBidAskData {

    public int reqId;
    public long time;
    public double bidPrice;
    public double askPrice;
    public Decimal bidSize;
    public Decimal askSize;
    public TickAttribBidAsk tickAttribBidAsk;


    public TickByTickBidAskData(int reqId,
                                long time,
                                double bidPrice,
                                double askPrice,
                                Decimal bidSize,
                                Decimal askSize,
                                TickAttribBidAsk tickAttribBidAsk) {
        this.reqId = reqId;
        this.time = time;
        this.bidPrice = bidPrice;
        this.askPrice = askPrice;
        this.bidSize = bidSize;
        this.askSize = askSize;
        this.tickAttribBidAsk = tickAttribBidAsk;
    }
}

File: 3_OrderSystem/KotlinEWrapperBackend/src/main/kotlin/com/seagosoft/tws/wrapper/structs/TickByTickBidAskData.kt
package com.seagosoft.tws.wrapper.structs

import com.ib.client.Decimal
import com.ib.client.TickAttribBidAsk


class TickByTickBidAskData(
    var reqId: Int,
    var time: Long,
    var bidPrice: Double,
    var askPrice: Double,
    var bidSize: Decimal,
    var askSize: Decimal,
    var tickAttribBidAsk: TickAttribBidAsk
)